spec/enumerable_spec.rb

```diff
@@ -0,0 +1,85 @@
+# spec/enumerables_spec.rb
+
+require_relative '../my_methods.rb'
+
+RSpec.describe Enumerable do
+  let(:test1) { [5, 4, 2] }
+  let(:test2) { [3] }
+  let(:test3) { [nil, true, 10] }
+  let(:test4) { [1, 2, 3, 1, 2, 3, 4, 1, 2, 3] }
+
+  describe '#my_each' do
+    it 'should interate through an array' do
+      expect(test1.my_each { |x| p x }).to eql(test1.each { |x| p x })
+      expect(test1.my_each { |x| p x * 2 }).to eql(test1.each { |x| p x * 2 })
+      expect(test1.my_each { |x| p x * 4 }).to eql(test1.each { |x| p x * 4 })
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'should interate through an array and its index' do
+      expect(test1.my_each_with_index { |x, y| x + y }).to eql(test1.each_with_index { |x, y| x + y })
+      expect(test1.my_each_with_index { |x, y| x * y }).to eql(test1.each_with_index { |x, y| x * y })
+      expect(test1.my_each_with_index { |x, y| x - y }).to eql(test1.each_with_index { |x, y| x - y })
+    end
+  end
+
+  describe '#my_select' do
+    it 'should create a new array based on the conditions passed at the block' do
+      expect(test2.my_select { |x| x == 1 }).to eql([])
+      expect(test3.my_select { |x| x == true }).to eql([true])
+      expect(test4.my_select { |x| x == 1 }).to eql([1, 1, 1])
+      expect(test4.my_select { |x| x > 3 }).to eql([4])
+    end
+  end
+
+  describe '#my_all?' do
+    it 'should check when the given condition is true for the itens in the array' do
+      expect(test1.my_all? { |x| x == 1 }).to eql(false)
+      expect(test1.my_all? { |x| x > 0 }).to eql(true)
+      expect(test1.my_all? { |x| x == true }).to eql(false)
+    end
+  end
+
+  describe '#my_any?' do
+    it 'should check whether the given condition is true for every item in the array' do
+      expect(test3.my_any? { |x| x == true }).to eql(true)
+      expect(test4.my_any? { |x| x == true }).to eql(false)
+      expect(test4.my_any? { |x| x > 0 }).to eql(true)
```

Comment:
You should have only one `it` block per `expected` behavior ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png)  from  Pathwright).